Source/WebCore/worklets/WorkletGlobalScope.cpp

```diff
@@ -126,13 +126,17 @@ URL WorkletGlobalScope::completeURL(const String& url, ForceUTF8) const
 
 void WorkletGlobalScope::logExceptionToConsole(const String& errorMessage, const String& sourceURL, int lineNumber, int columnNumber, RefPtr<ScriptCallStack>&& stack)
 {
+    if (UNLIKELY(settingsValues().logsPageMessagesToSystemConsoleEnabled))
+        PageConsoleClient::logMessageToSystemConsole(*makeUnique<Inspector::ConsoleMessage>(MessageSource::JS, MessageType::Log, MessageLevel::Error, errorMessage, sourceURL, lineNumber, columnNumber));
     if (!m_document || isJSExecutionForbidden())
         return;
     m_document->logExceptionToConsole(errorMessage, sourceURL, lineNumber, columnNumber, WTFMove(stack));
 }
 
 void WorkletGlobalScope::addConsoleMessage(std::unique_ptr<Inspector::ConsoleMessage>&& message)
 {
+    if (UNLIKELY(settingsValues().logsPageMessagesToSystemConsoleEnabled && message))
+        PageConsoleClient::logMessageToSystemConsole(*message);
```

Comment:
Can we check sessionID like we do in WorkerGlobalScope::addConsoleMessage?

Reply:
Sure.

Reply:
On second thought, that seems to always suppress the console messages. Is it possible that WorkletGlobalScope sessions are always considered ephemeral?

Reply:
I do not think so, SessionID should be process wide and should not be ephemeral for regular web pages.